### PR TITLE
Update zhtlc methods for next API release

### PR DESCRIPTION
**Pull request by smk762**
Closes: https://github.com/KomodoPlatform/atomicDEX-Desktop/issues/2046
To Test:
- Activate ARRR / ZOMBIE
- Do an ARRR/ZOMBIE transaction

### CODE REVIEW

**Review by cipig — APPROVED** [review]
image

**Comment by tonymorony** [comment]
@smk762 
```
Conflicting files
CMakeLists.txt

```

**Review by cipig — DISMISSED** [review]

**Review by cipig — APPROVED** [review]

**Comment by smk762** [comment]
Latest API release still not using these methods new yet.

**Review by cipig — APPROVED** [review]

**Comment by smk762** [comment]
Local testing with latest API:
- able to send without problem
- crashed when trying to swap
```
]
[13:49:26] [error] [main.prerequisites.hpp:113] [904690]: stacktrace:  0# boost::stacktrace::safe_dump_to(char const*) at /home/smk762/GITHUB/KP/atomicDEX-Desktop/build/../ci_tools_atomic_dex/vcpkg-repo/installed/x64-linux/include/boost/stacktrace/safe_dump_to.hpp:124
 1# 0x00007FCDDA1AD090 in /lib/x86_64-linux-gnu/libc.so.6
 2# gsignal in /lib/x86_64-linux-gnu/libc.so.6
 3# abort in /lib/x86_64-linux-gnu/libc.so.6
 4# 0x00007FCDDA18C729 in /lib/x86_64-linux-gnu/libc.so.6
 5# 0x00007FCDDA19DFD6 in /lib/x86_64-linux-gnu/libc.so.6
 6# atomic_dex::orderbook_model::reset_orderbook(std::vector<atomic_dex::mm2::order_contents, std::allocator<atomic_dex::mm2::order_contents> > const&) at /home/smk762/GITHUB/KP/atomicDEX-Desktop/build/../src/core/atomicdex/models/qt.orderbook.model.cpp:385
 7# atomic_dex::qt_orderbook_wrapper::refresh_orderbook(atomic_dex::mm2::orderbook_answer) at /home/smk762/GITHUB/KP/atomicDEX-Desktop/build/../src/core/atomicdex/pages/widgets/dex/qt.orderbook.cpp:94
 8# atomic_dex::trading_page::process_action() at /home/smk762/GITHUB/KP/atomicDEX-Desktop/build/../src/core/atomicdex/pages/qt.trading.page.cpp:492
 9# atomic_dex::application::tick() at /home/smk762/GITHUB/KP/atomicDEX-Desktop/build/../src/app/app.cpp:?
10# QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, void (atomic_dex::application::*)()>::call(void (atomic_dex::application::*)(), atomic_dex::application*, void**) at /home/smk762/Qt/5.15.2/gcc_64/include/QtCore/qobjectdefs_impl.h:152
11# void QtPrivate::FunctionPointer<void (atomic_dex::application::*)()>::call<QtPrivate::List<>, void>(void (atomic_dex::application::*)(), atomic_dex::application*, void**) at /home/smk762/Qt/5.15.2/gcc_64/include/QtCore/qobjectdefs_impl.h:186
12# QtPrivate::QSlotObject<void (atomic_dex::application::*)(), QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) at /home/smk762/Qt/5.15.2/gcc_64/include/QtCore/qobjectdefs_impl.h:419
13# 0x00007FCDDB8A8F30 in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
14# QTimer::timeout(QTimer::QPrivateSignal) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
15# QTimer::timerEvent(QTimerEvent*) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
16# QObject::event(QEvent*) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
17# QApplicationPrivate::notify_helper(QObject*, QEvent*) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Widgets.so.5
18# QApplication::notify(QObject*, QEvent*) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Widgets.so.5
19# QCoreApplication::notifyInternal2(QObject*, QEvent*) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
20# QTimerInfoList::activateTimers() in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
21# 0x00007FCDDB8CBE99 in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
22# g_main_context_dispatch in /lib/x86_64-linux-gnu/libglib-2.0.so.0
23# 0x00007FCDD708D400 in /lib/x86_64-linux-gnu/libglib-2.0.so.0
24# g_main_context_iteration in /lib/x86_64-linux-gnu/libglib-2.0.so.0
25# QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
26# QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
27# QCoreApplication::exec() in /home/smk762/Qt/5.15.2/gcc_64/lib/libQt5Core.so.5
28# run_app(int, char**) at /home/smk762/GITHUB/KP/atomicDEX-Desktop/build/../src/app/main.prerequisites.hpp:517
29# main at /home/smk762/GITHUB/KP/atomicDEX-Desktop/build/../src/app/main.cpp:38
30# __libc_start_main in /lib/x86_64-linux-gnu/libc.so.6
31# _start in ./atomicdex-desktop
```

**Comment by cipig** [comment]
had a crash too yesterday when tried to start a swap, but i don't think it's related to this... the swap was KMD->QTUM, nothing with ARRR

```
terminate called after throwing an instance of 'std::out_of_range'
  what():  vector::_M_range_check: __n (which is 234) >= this->size() (which is 0)
```

log shows
```
[19:34:34] [error] [main.prerequisites.hpp:113] [3207840]: stacktrace:  0# signal_handler(int) at main.cpp:?
 1# 0x00007FDD3D263D60 in /lib/x86_64-linux-gnu/libc.so.6
 2# gsignal in /lib/x86_64-linux-gnu/libc.so.6
 3# abort in /lib/x86_64-linux-gnu/libc.so.6
 4# 0x00007FDD41ECD7EC in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 5# 0x00007FDD41ED8966 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 6# 0x00007FDD41ED89D1 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 7# 0x00007FDD41ED8C65 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 8# std::__throw_out_of_range_fmt(char const*, ...) in /home/florian/Tools/AntaraAtomicDexAppDir/usr/bin/../lib/libQt5Positioning.so.5
 9# atomic_dex::orderbook_model::data(QModelIndex const&, int) const in AntaraAtomicDexAppDir/usr/bin/atomicdex-desktop
10# QSortFilterProxyModel::data(QModelIndex const&, int) const in /home/florian/Tools/AntaraAtomicDexAppDir/usr/bin/../lib/libQt5Core.so.5
11# 0x00007FDD4083DB9A in /home/florian/Tools/AntaraAtomicDexAppDir/usr/bin/../lib/libQt5QmlModels.so.5
12# 0x00007FDD4019F9CE in /home/florian/Tools/AntaraAtomicDexAppDir/usr/bin/../lib/libQt5Qml.so.5
13# QV4::QObjectWrapper::getQmlProperty(QV4::ExecutionEngine*, QQmlContextData*, QObject*, QV4::String*, QV4::QObjectWrapper::RevisionMode, bool*, QQmlPropertyData**) in /home/florian/Tools/AntaraAtomicDexAppDir/usr/bin/../lib/libQt5Qml.so.5
14# QV4::QQmlContextWrapper::lookupInParentContextHierarchy(QV4::Lookup*, QV4::ExecutionEngine*, QV4::Value*) in /home/florian/Tools/AntaraAtomicDexAppDir/usr/bin/../lib/libQt5Qml.so.5
15# 0x00007FDD1EFB4771
```

and lots of
```
[19:34:34] [error] [mm2.service.cpp:2511] [3207871]: pplx task error: Failed to read HTTP status line from: fetch_single_balance
```
before for different coins

followed by 2
```
[19:34:34] [error] [mm2.service.cpp:2511] [3207856]: pplx task error: Failed to read HTTP status line from: batch_fetch_orders_and_swap
```

guess it happens when you start a swap while ADEX is updating balances
btw, do we need to update balances manually in a loop all the time? couldn't we just subscribe to block headers https://electrumx-spesmilo.readthedocs.io/en/latest/protocol-methods.html#blockchain-headers-subscribe or scripthash https://electrumx-spesmilo.readthedocs.io/en/latest/protocol-methods.html#blockchain-scripthash-subscribe ?

File: src/core/atomicdex/services/mm2/mm2.service.cpp [hunk 1]
```diff
@@ -2095,8 +2134,8 @@ namespace atomic_dex
         if (it == m_balance_informations.cend())
         {
             ec = dextop_error::unknown_ticker;
-            SPDLOG_INFO("Invalid Ticker {}", ticker);
-            return "Invalid Ticker";
+            SPDLOG_INFO("Ticker {} is invalid or not yet enabled", ticker);
+            return "Ticker not yet enabled";
```
Comment (cipig) [inline src/core/atomicdex/services/mm2/mm2.service.cpp:2138]: is "Invalid Ticker" check here https://github.com/KomodoPlatform/atomicDEX-Desktop/blob/dev/atomic_defi_design/Dex/Settings/RecoverSeedModal.qml#L375 related? If so, needs to be changed too.

Comment (smk762) [reply src/core/atomicdex/services/mm2/mm2.service.cpp:2138]: Checking that area now to suppress balance/history requests until zhtlc completely activated.

**Comment by smk762** [comment]
closing in favour of https://github.com/KomodoPlatform/atomicDEX-Desktop/pull/2230

**Pull request closed without merging**